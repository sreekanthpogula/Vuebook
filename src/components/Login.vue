<template>
    <v-app>
       <v-main>
          <v-container fluid fill-height>
             <v-layout align-center justify-center>
                <v-flex xs12 sm8 md4>
                   <v-card class="elevation-12">
                      <v-toolbar dark color="primary">
                         <v-toolbar-title>Vuebook {{isRegister ? stateObj.register.name : stateObj.login.name}}</v-toolbar-title>
                      </v-toolbar>
                      <v-card-text>
                      <form ref="form" @submit.prevent="isRegister ? register() : login()">
                        <v-text-field label="Username" prepend-icon="mdi-account-circle"/>
                            <v-text-field
                               v-model="password"
                               name="password"
                               label="Password"
                               :type="showPassword ? 'text' : 'password'"
                                prepend-icon="mdi-lock"
                                :append-icon="showPassword ? 'mdi-eye' : 'mdi-eye-off'"
                                @click:append="showPassword = !showPassword">
                            </v-text-field>
                            <v-text-field v-if="isRegister"
                               v-model="confirmPassword"
                               name="confirmPassword"
                               label="Confirm Password"
                               type="password"
                               placeholder="confirm password"
                               required
                            ></v-text-field>
                            <div class="red--text"> {{errorMessage}}</div>
                            <v-btn type="submit" class="mt-4" color="primary" value="log in">{{isRegister ? stateObj.register.name : stateObj.login.name}}</v-btn>
                            <div class="grey--text mt-4" v-on:click="isRegister = !isRegister;">
                               {{toggleMessage}}  
                            </div>
                       </form>
                      </v-card-text>
                   </v-card>
                 
                </v-flex>
             </v-layout>
          </v-container>
       </v-main>
    </v-app>
 </template>
 
 <script>
 export default {
   name: "App",
   data() {
     return {
       username: "",
       password: "",
       confirmPassword: "",
       isRegister : false,
       errorMessage: "",
       stateObj: {
          register :{
             name: 'Register',
             message: 'Aleady have an Acoount? login.'
          },
          login : {
             name: 'Login',
             message: 'Register'
          }
       }
     };
   },
   methods: {
     login() {
       const { username } = this;
       this.$router.replace({ name: "dashboard", params: { username: username } });
     },
     register() {
        if(this.password == this.confirmPassword){
           this.isRegister = false;
           this.errorMessage = "";
           this.$refs.form.reset();
        }
        else {
          this.errorMessage = "password did not match"
        }
     }
   },
       computed: {
        toggleMessage : function() { 
           return this.isRegister ? this.stateObj.register.message : this.stateObj.login.message }
     }
 };
 </script>