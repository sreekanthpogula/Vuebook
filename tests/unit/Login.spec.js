import Vuetify from 'vuetify'

// Components
import Login from '@/components/Login.vue'
import { createLocalVue, mount } from '@vue/test-utils'
describe('Login.vue', () => {
    const localVue = createLocalVue()
    let vuetify
  
    beforeEach(() => {
      vuetify = new Vuetify()
    })
  
    const mountFunction = options => {
      return mount(Login, {
        localVue,
        vuetify,
        ...options,
      })
    }
  
    it('should have a custom title and match snapshot', () => {
      const wrapper = mountFunction({
        propsData: { title: 'Foobar' },
      })
  
      expect(wrapper.html()).toMatchSnapshot()
    })
  })