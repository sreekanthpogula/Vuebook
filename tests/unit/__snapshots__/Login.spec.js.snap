// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login.vue should have a custom title and match snapshot 1`] = `
<v-app title="Foobar">
  <v-main>
    <v-container fluid="" fill-height="">
      <v-layout align-center="" justify-center="">
        <v-flex xs12="" sm8="" md4="">
          <v-card class="elevation-12">
            <v-toolbar dark="" color="primary">
              <v-toolbar-title>Vuebook Register</v-toolbar-title>
            </v-toolbar>
            <v-card-text>
              <form>
                <v-text-field label="Username" prepend-icon="mdi-account-circle"></v-text-field>
                <v-text-field name="password" label="Password" type="password" prepend-icon="mdi-lock" append-icon="mdi-eye-off"></v-text-field>
                <v-text-field name="confirmPassword" label="Confirm Password" type="password" placeholder="confirm password" required="" prepend-icon="mdi-lock" append-icon="mdi-eye-off">
                  &gt;</v-text-field>
                <div class="red--text"> </div>
                <v-btn type="submit" color="primary" value="log in" class="mt-4">Register</v-btn>
                <div class="grey--text mt-4">
                  Aleady have an Acoount? login.
                </div>
              </form>
            </v-card-text>
          </v-card>
        </v-flex>
      </v-layout>
    </v-container>
  </v-main>
</v-app>
`;
